packages/backend/src/controllers/authentication.ts

```diff
@@ -73,6 +76,150 @@ export const getOpenIdUserFromRequest = ({
+        lightdashConfig.siteUrl,
+    ).href;
+
+    const client = new oktaIssuer.Client({
+        client_id: lightdashConfig.auth.okta.oauth2ClientId ?? '',
+        client_secret: lightdashConfig.auth.okta.oauth2ClientSecret,
+        redirect_uris: [redirectUri],
+        response_types: ['code'],
+    });
+
+    return client;
+};
+
+export class OpenIDClientOktaStrategy extends Strategy {
+    private clientPromise: Promise<Client>;
+
+    constructor() {
+        super();
+        this.clientPromise = setupClient();
+    }
+
+    async authenticate(req: Request) {
+        try {
+            const client = await this.clientPromise;
+
+            const redirectUri = new URL(
+                `/api/v1${lightdashConfig.auth.okta.callbackPath}`,
+                lightdashConfig.siteUrl,
+            ).href;
+
+            const params = client.callbackParams(req);
+            const tokenSet = await client.callback(redirectUri, params, {
+                code_verifier: req.session.oauth?.codeVerifier,
+                state: req.session.oauth?.state,
+            });
+
+            const userInfo = tokenSet.access_token
+                ? await client.userinfo(tokenSet.access_token)
+                : undefined;
+
+            if (!userInfo) return this.fail(401);
+
+            try {
+                const { inviteCode } = req.session.oauth || {};
+                req.session.oauth = {};
+
+                const user = await userService.loginWithOpenId(
+                    createOpenIdUser(
+                        userInfo,
+                        undefined,
+                        tokenSet.claims().iss,
+                        OpenIdIdentityIssuerType.OKTA,
+                    ),
+                    req.user,
+                    inviteCode,
+                );
+                return this.success(user);
+            } catch (e) {
+                if (e instanceof LightdashError) {
+                    return this.fail({ message: e.message }, 401);
+                }
+                Logger.warn(`Unexpected error while authorizing user: ${e}`);
+                return this.error(e);
+            }
+        } catch (err) {
+            this.error(err);
+        }
+        return this.fail(401);
+    }
+}
+
+export const initiateOktaOpenIdLogin: RequestHandler = async (req, res) => {
+    const client = await setupClient();
+
+    const redirectUri = new URL(
+        `/api/v1${lightdashConfig.auth.okta.callbackPath}`,
+        lightdashConfig.siteUrl,
+    ).href;
+
+    const state = generators.state();
+    const codeVerifier = generators.codeVerifier();
+    const codeChallenge = generators.codeChallenge(codeVerifier);
+
+    const authorizationUrl = client.authorizationUrl({
+        redirect_uri: redirectUri,
+        response_type: 'code',
+        scope: 'openid profile email',
+        code_challenge: codeChallenge,
+        code_challenge_method: 'S256',
+        state,
+    });
+
+    req.session.oauth = {
```

Comment:
does the redirect work correctly btw? Could you test accessing a dashboard link directly when logged out then complete the okta sign in and you should end up on the dashboard

Reply:
Checking this now!

Reply:
It works! Copied dashboard link, logged out, input the dashboard link, sign in, automatic redirect to the dashboard